universal_silabs_flasher/flasher.py

```diff
@@ -91,17 +94,33 @@ def __init__(
         self.app_baudrate: int | None = None
         self.bootloader_baudrate: int | None = None
 
+        self._reset_target:ResetTarget | None = \
+            ResetTarget(reset_method) if reset_method else None
+
     async def enter_yellow_bootloader(self):
         _LOGGER.info("Triggering Yellow bootloader")
 
         await send_gpio_pattern(
+            chip_id=0,
             pin_states={
                 24: [True, False, False],
                 25: [True, False, True],
             },
             toggle_delay=0.1,
         )
 
+    async def enter_ihost_bootloader(self):
```

Comment:
I think it would be best to break out the bootloader patterns into a dictionary or something, keyed by the device. Or maybe we allow specifying the pattern as command line arguments?

Reply:
Sure I can breakout the bootloader config to a dict. 
Not sure specifying via cli options would be very user friendly, having to remember which pins are required etc.

Reply:
I have spilt out the gpio pin configs for bootloader reset to a dict.
These have been rebased on top of PR#50